pkg/resources/connection_acceptance_test.go

```diff
@@ -0,0 +1,421 @@
+	_ = testenvs.GetOrSkipTest(t, testenvs.TestFailoverGroups)
+
+	id := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	accountId := acc.TestClient().Account.GetAccountIdentifier(t)
+	primaryConnectionAsExternalId := sdk.NewExternalObjectIdentifier(accountId, id)
+
+	connectionModel := model.Connection("t", id.Name()).WithComment("config comment")
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.Connection),
+		Steps: []resource.TestStep{
+			// create
+			{
+				Config: config.FromModel(t, connectionModel),
+				Check: resource.ComposeTestCheckFunc(
+					assert.AssertThat(t,
+						resourceassert.ConnectionResource(t, connectionModel.ResourceReference()).
+							HasNameString(id.Name()).
+							HasFullyQualifiedNameString(id.FullyQualifiedName()).
+							HasNoAsReplicaOf().
+							HasNoEnableFailoverToAccounts().
+							HasNoIsPrimary().
+							HasCommentString("config comment"),
+
+						resourceshowoutputassert.ConnectionShowOutput(t, connectionModel.ResourceReference()).
+							HasName(id.Name()).
+							HasSnowflakeRegion(acc.TestClient().Context.CurrentRegion(t)).
+							HasAccountLocator(acc.TestClient().GetAccountLocator()).
+							HasAccountName(accountId.AccountName()).
+							HasOrganizationName(accountId.OrganizationName()).
+							HasComment("config comment").
+							HasIsPrimary(true).
+							HasPrimaryIdentifier(primaryConnectionAsExternalId).
+							HasFailoverAllowedToAccounts(accountId),
+					),
+				),
+			},
+			// change comment externally
+			{
+				PreConfig: func() {
+					acc.TestClient().Connection.Alter(t, sdk.NewAlterConnectionRequest(id).
+						WithSet(*sdk.NewSetConnectionRequest().
+							WithComment("external comment")))
+				},
+				Config: config.FromModel(t, connectionModel),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(connectionModel.ResourceReference(), plancheck.ResourceActionUpdate),
+						planchecks.ExpectChange(connectionModel.ResourceReference(), "comment", tfjson.ActionUpdate, sdk.String("external comment"), sdk.String("config comment")),
+					},
+				},
+				Check: resource.ComposeTestCheckFunc(
+					assert.AssertThat(t,
+						resourceassert.ConnectionResource(t, connectionModel.ResourceReference()).
+							HasCommentString("config comment"),
+						resourceshowoutputassert.ConnectionShowOutput(t, connectionModel.ResourceReference()).
+							HasComment("config comment"),
+					),
+				),
+			},
+		},
+	})
+}
+
+// TODO: [SNOW-1763442]
+/*
+func TestAcc_Connection_ExternalChangeToIsPrimary(t *testing.T) {
+	// TODO: [SNOW-1002023]: Unskip; Business Critical Snowflake Edition needed
```

Comment:
We should uncomment this test - it will be skipped anyway.

Reply:
will do